app.go

```diff
@@ -323,6 +324,13 @@ type Config struct {
 	// Default: json.Unmarshal
 	JSONDecoder utils.JSONUnmarshal `json:"-"`
 
+	// When set by an external client of Fiber it will use the provided implementation of a
+	// RegexMatcher
+	//
+	// Allowing for flexibility in using another regex library for matching
+	// Default: regexp.MatchString
+	RegexMatcher utils.RegexMatch `json:"-"`
```

Comment:
please set the "regexp.MatchString" as default matcher, then we don't need optional parameters in the match functionality anymore

Reply:
What about adding private var? We can assign app config to that variable. It'd be easier to get it outsuide of app methods.

Reply:
just do it like this 
https://github.com/gofiber/fiber/blob/e5eb8d3c98b33d8047cdc27bc165330f4f8b5312/app.go#L506-L510